hamaddr/src/lib.rs

```diff
@@ -0,0 +1,338 @@
+use anyhow::{bail, format_err};
+use std::convert::TryInto;
+use std::fmt;
+
+type Result<T> = std::result::Result<T, anyhow::Error>;
+
+/// Eui48 represents an EUI48 MAC address.
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Hash)]
+pub struct Eui48(pub [u8; 6]);
+
+impl Eui48 {
+    /// Creates a new Eui48 from the given six octets.
+    pub const fn new(addr: [u8; 6]) -> Eui48 {
+        Eui48(addr)
+    }
+}
+/// Converts an Eui48 into a HamAddr
+impl Into<HamAddr> for Eui48 {
```

Comment:
I think this may need to be a `TryInto`, because the EUI48 may not translate properly into a `HamAddr`.

Reply:
Interface changed, with a TODO to implement proper error checking.